extra/checksums/multi/multi.factor

```diff
@@ -0,0 +1,31 @@
+! Copyright (C) 2018 Alexander Ilin.
+! See http://factorcode.org/license.txt for BSD license.
+USING: accessors checksums checksums.common destructors fry kernel
+sequences ;
+IN: checksums.multi
+
+TUPLE: multi-checksum list ;
+INSTANCE: multi-checksum block-checksum
+C: <multi-checksum> multi-checksum
+
+TUPLE: multi-state < disposable states results ;
+
+M: multi-checksum initialize-checksum-state
+    list>> [ initialize-checksum-state ] V{ } map-as
```

Comment:
Couple things here:

1) Don't like the name ``list``, maybe ``checksums``.

2) Block checksums create ``block-checksum-state`` that have ``block-size``.  That's used to split the blocks into groups of the correct size when ``checksum-bytes`` is called.

So, I would probably make ``multi-state`` be a ``block-checksum-state``, and assert that all checksum-states that are created have the same ``block-size`` (otherwise they aren't going to work properly).  For example, ``md5`` might have 64-byte blocks, but ``metrohash-64`` has 32-byte blocks.

Reply:
You also, might want to implement ``checksum-bytes``:

```factor
M: multi-checksum checksum-bytes
    list>> [ checksum-bytes ] with map ;
```

Reply:
Actually, nevermind about the ``block-checksum-state``.  Your delegation of ``add-checksum-bytes`` is just fine.

Reply:
And also, if you're a ``block-checksum``, then you inherit ``checksum-bytes`` which just defers to your ``checksum-state`` implementation.

Reply:
Renamed `list` to `checksums`.